static/js/components/EducationDialog.js

```diff
@@ -23,11 +26,16 @@ export default class EducationDialog extends ProfileFormFields {
   }
   educationLevelLabels: Object;
 
-  static propTypes = {
-    open:           React.PropTypes.bool,
-    onClose:        React.PropTypes.func,
-    onSave:         React.PropTypes.func,
-    showLevelForm:  React.PropTypes.bool,
+  props: {
+    ui:                           UIState,
+    setEducationDialogVisibility: () => void,
+    setEducationDegreeLevel:      () => void,
+    setEducationDialogIndex:      () => void,
+    clearProfileEdit:             () => void,
+    saveProfile:                  BoundSaveProfile,
```

Comment:
to the first paragraph, is it possible to be more vague and tell flow that the type is just `Function` and be done with it? maybe it

to the second paragraph, my comments still hold. `boolean` makes sense as a type. `Object` makes sense as a type. `UIValidator` makes sense as a type, in that it suggests that i'm probably expecting a function for this variable that performs some validation. `BoundSaveProfile` makes some sense as the name of a _function_, but IMO it does not make sense as the name of a _type_. `ProfileSaveFunction`, et al, tells me what to expect here more accurately. 

as for `Bound`, we can't have any expectation at the _type_ level that this function has been bound to something before it got here. all the _type_ can say is that it's a function that accepts certain arguments and returns a value of a certain type. the "bound" part of the name is an implementation detail, and i don't think we should be insisting on that in the name of the type